crates/sui-core/src/authority.rs

```diff
@@ -1439,17 +1425,45 @@ impl AuthorityState {
         }
 
         let scheduling_source = execution_env.scheduling_source;
-        let (transaction_outputs, timings, execution_error_opt) = self
-            .process_certificate(
-                &tx_guard,
-                &execution_guard,
-                certificate,
-                execution_env,
-                epoch_store,
+        let mysticeti_fp_outputs = if epoch_store.protocol_config().mysticeti_fastpath()
+            && !certificate.is_consensus_tx()
+            && scheduling_source == SchedulingSource::NonFastPath
```

Comment:
Maybe we don't check for `scheduling_source == SchedulingSource::NonFastPath` here? Basically there are 4 cases for executing owned object txns:
1. Executing the txn first time from FP.
2. Executing the txn first time from consensus output / checkpoint.
3. Duplicated execution from FP, where result is already in FP cache.
4. Duplicated execution from consensus output / checkpoint, where result is already in FP cache.

I assume no execution can happen after the txn has committed. I think 1/2/4 are handled correctly. 3 is correct but it seems duplicated execution will happen instead of reusing the results from FP cache?